api/envoy/config/filter/http/transformation/v2/transformation_filter.proto

```diff
@@ -7,6 +7,18 @@ option java_outer_classname = "TransformationFilterProto";
 option java_multiple_files = true;
 option go_package = "transformation";
 
+import "validate/validate.proto";
+
+import "envoy/api/v2/route/route.proto";
+import "envoy/type/matcher/string.proto";
+
+message FilterTransformations {
+  // specific headers to match
+  repeated envoy.api.v2.route.HeaderMatcher header_matchers = 2;
```

Comment:
i'm reasonabliy sure that the api needs to be 
```
message Transforms {
envoy.api.v2.route.HeaderMatcher header_matchers = 2;
RouteTransformations route_transformations = 3;
}
repeated Transforms = 1;
```

similar to how rules work in the json filter:
https://github.com/envoyproxy/envoy/blob/24f3b527b461f0d33deae4c36cb9456fa0c15ae2/api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto#L458